src/pe/debug.rs

```diff
@@ -35,14 +39,36 @@ impl<'a> DebugData<'a> {
         file_alignment: u32,
         opts: &options::ParseOptions,
     ) -> error::Result<Self> {
-        let image_debug_directory =
-            ImageDebugDirectory::parse_with_opts(bytes, dd, sections, file_alignment, opts)?;
-        let codeview_pdb70_debug_info =
-            CodeviewPDB70DebugInfo::parse_with_opts(bytes, &image_debug_directory, opts)?;
-
+        let rva = dd.virtual_address as usize;
+        let mut offset =
+            utils::find_offset(rva, sections, file_alignment, opts).ok_or_else(|| {
+                error::Error::Malformed(format!(
+                    "Cannot map ImageDebugDirectory rva {:#x} into offset",
+                    rva
+                ))
+            })?;
+
+        let len = dd.size as usize;
+        let sizeof_directory = ImageDebugDirectory::size_with(&scroll::LE);
+        let num_entries = len / sizeof_directory;
+
+        let mut entries = (0..num_entries)
+            .map(|_| bytes.gread_with(&mut offset, scroll::LE))
+            .collect::<Result<Vec<ImageDebugDirectory>, _>>()?;
+
+        // find the debug directory that references the codeview record
+        for (idx, idd) in entries.iter().enumerate() {
```

Comment:
I know the original code just tries parsing `CodeviewPDB70DebugInfo` directly, but you should additionally be able to look at the `data_type` field to see if it is `IMAGE_DEBUG_TYPE_CODEVIEW` per [the MSDN docs](https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_debug_directory).